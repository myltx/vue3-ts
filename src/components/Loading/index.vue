
<script setup lang="ts">
const props = defineProps({
  msg: {
    type: String,
    default: 'hello'
  }
})
</script>
<template>
	{{ msg }}
	<div>
		loading
	</div>
</template>

<style scoped>

</style>
