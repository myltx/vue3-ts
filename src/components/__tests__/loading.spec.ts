import {describe, it, expect} from 'vitest'
import {mount} from '@vue/test-utils'
import Loading from '@/components/Loading/index.vue'

describe('Loading component', (
) => {
  it('renders correctly', () => {
    expect(Loading).toBeTruthy()
    const wrapper = mount(Loading, {
      props: {
			  msg: '123'
      }
    })
    expect(wrapper.text()).toContain('123')
    // expect(wrapper.html()).toMatchSnapshot()
  })
})
