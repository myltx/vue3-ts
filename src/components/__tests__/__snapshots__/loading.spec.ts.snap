// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading component > renders correctly 1`] = `"<div> loading </div>"`;
